File: api/server-common/src/main/java/su/plo/voice/api/server/audio/source/BaseServerSourceManager.java
package su.plo.voice.api.server.audio.source;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;
import su.plo.voice.api.audio.source.AudioSourceManager;
import su.plo.voice.api.server.audio.line.ServerSourceLine;

import java.util.UUID;

public interface BaseServerSourceManager extends AudioSourceManager<ServerAudioSource<?>> {

    @NotNull ServerDirectSource createDirectSource(@NotNull Object addonObject,
                                                   @NotNull ServerSourceLine line,
                                                   @Nullable String codec,
                                                   boolean stereo);

    void remove(@NotNull UUID sourceId);

    void remove(@NotNull ServerAudioSource<?> source);
}

File: api/server-common/src/main/kotlin/su/plo/voice/api/server/audio/source/BaseServerSourceManager.kt
package su.plo.voice.api.server.audio.source

import su.plo.voice.api.audio.source.AudioSourceManager
import su.plo.voice.api.server.audio.line.ServerSourceLine
import su.plo.voice.proto.data.audio.codec.CodecInfo
import su.plo.voice.proto.data.audio.codec.opus.OpusDecoderInfo
import java.util.*

interface BaseServerSourceManager : AudioSourceManager<ServerAudioSource<*>> {

    fun createDirectSource(
        addonObject: Any,
        line: ServerSourceLine,
        stereo: Boolean = false,
        decoderInfo: CodecInfo? = OpusDecoderInfo()
    ): ServerDirectSource

    fun remove(sourceId: UUID)

    fun remove(source: ServerAudioSource<*>)
}
